app/lib/locationValidator.js

```diff
@@ -2,20 +2,22 @@ const Postcode = require('postcode');
 
 function validateLocation(location) {
   let errorMessage = null;
+  let locationToReturn = location;
 
   if (!location) {
     errorMessage = 'A valid postcode is required to progress';
   } else {
-    const postcode = new Postcode(location);
+    locationToReturn = location.trim();
+    const postcode = new Postcode(locationToReturn);
 
-    if (!postcode.valid() && !Postcode.validOutcode(location)) {
-      errorMessage = `${location} is not a valid postcode, please try again`;
+    if (!postcode.valid() && !Postcode.validOutcode(locationToReturn)) {
+      errorMessage = `${locationToReturn} is not a valid postcode, please try again`;
     }
   }
 
   return {
     errorMessage,
-    input: location,
+    input: locationToReturn,
```

Comment:
I can see why you need to do the let in line 5 to prevent side effects from references. Would it be better to make the intent explicit through a function?

e.g.  `function cloneString(string) { return cloneString = string; )`
...
`input: cloneString(location)`